File: ios/Plugin/MetadataConfigPlugin.swift
import Foundation
import Capacitor

@objc(MetadataConfigPlugin)
public class MetadataConfigPlugin: CAPPlugin {
    private let implementation = MetadataConfig()

    @objc func read(_ call: CAPPluginCall) {
        let key = call.getString("key") ?? ""
        
        let value = implementation.read(key)
        if (value == nil) {
            call.reject("value for \(key) not found. Did you add the key/value to Info.plist?")
        }
        call.resolve([
            "value": value!
        ])
    }
}

File: android/src/main/java/com/getcapacitor/community/metadataconfig/MetadataConfigPlugin.kt
package com.getcapacitor.community.metadataconfig

import com.getcapacitor.JSObject
import com.getcapacitor.Plugin
import com.getcapacitor.PluginCall
import com.getcapacitor.PluginMethod
import com.getcapacitor.annotation.CapacitorPlugin

@CapacitorPlugin(name = "MetadataConfig")
class MetadataConfigPlugin : Plugin() {
    private val implementation = MetadataConfig()
    @PluginMethod
    fun read(call: PluginCall) {
        val key = call.getString("key")

        val ret = JSObject()
        val value = implementation.read(context, key!!)
        if (value == null) {
            call.reject("value for key $key not found. Did you add it to AndroidManifest.xml?")
        }
        ret.put("value", value)
        call.resolve(ret)
    }
}